sagemaker-debugger/tensorflow2/tensorflow2_keras_custom_container/src/tf_keras_resnet_byoc.py

```diff
@@ -39,58 +37,58 @@ def train(batch_size, epoch, model, hook):
     X_valid -= mean_image
     X_train /= 128.
     X_valid /= 128.
-
+    
+    # register hook to save the following scalar values
+    hook.save_scalar("epoch", epoch)
+    hook.save_scalar("batch_size", batch_size)
+    hook.save_scalar("train_steps_per_epoch", len(X_train)/batch_size)
+    hook.save_scalar("valid_steps_per_epoch", len(X_valid)/batch_size)
+    
     model.fit(X_train, Y_train,
               batch_size=batch_size,
               epochs=epoch,
               validation_data=(X_valid, Y_valid),
-              shuffle=True,
+              shuffle=False,
               # smdebug modification: Pass the hook as a Keras callback
               callbacks=[hook])
 
 
 def main():
     parser = argparse.ArgumentParser(description="Train resnet50 cifar10")
-    parser.add_argument("--batch_size", type=int, default=32)
-    parser.add_argument("--epoch", type=int, default=3)
+    parser.add_argument("--batch_size", type=int, default=50)
+    parser.add_argument("--epoch", type=int, default=15)
     parser.add_argument("--model_dir", type=str, default="./model_keras_resnet")
     parser.add_argument("--lr", type=float, default=0.001)
     parser.add_argument("--random_seed", type=bool, default=False)
-    parser.add_argument("--num_epochs", type=int, default=5, help="Number of epochs to train for")
-    parser.add_argument(
-        "--num_steps",
-        type=int,
-        help="Number of steps to train for. If this" "is passed, it overrides num_epochs",
-    )
-    parser.add_argument(
-        "--num_eval_steps",
-        type=int,
-        help="Number of steps to evaluate for. If this"
-             "is passed, it doesnt evaluate over the full eval set",
-    )
+    
     args = parser.parse_args()
 
     if args.random_seed:
         tf.random.set_seed(2)
         np.random.seed(2)
         random.seed(12)
 
-    model = ResNet50(weights=None, input_shape=(32,32,3), classes=10)
-
-    # smdebug modification:
-    # Create hook from the configuration provided through sagemaker python sdk.
-    # This configuration is provided in the form of a JSON file.
-    # Default JSON configuration file:
-    # {
-    #     "LocalPath": <path on device where tensors will be saved>
-    # }"
-    # Alternatively, you could pass custom debugger configuration (using DebuggerHookConfig)
-    # through SageMaker Estimator. For more information, https://github.com/aws/sagemaker-python-sdk/blob/master/doc/amazon_sagemaker_debugger.rst
-    hook = smd.KerasHook.create_from_json_file()
-    opt = tf.keras.optimizers.Adam(learning_rate=args.lr)
-    model.compile(loss='categorical_crossentropy',
-                  optimizer=opt,
-                  metrics=['accuracy'])
+        
+    mirrored_strategy = tf.distribute.MirroredStrategy()
```

Comment:
why moving this to mirrored strategy? apart from removing the command line args num_epochs, num_steps, and num_eval_steps, no other change was required in this script?